js/coinex.js

```diff
@@ -990,6 +1031,100 @@ module.exports = class coinex extends Exchange {
         return this.parseTrades (trades, market, since, limit);
     }
 
+    async transfer (code, amount, fromAccount, toAccount, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        const accountsByType = this.safeValue (this.options, 'accountsByType', {});
+        const fromId = this.safeNumber (accountsByType, fromAccount);
+        const toId = this.safeNumber (accountsByType, toAccount);
+        const request = {
+            'coin_type': currency['id'],
+            'amount': this.currencyToPrecision (code, amount),
+        };
+        let method = undefined;
+        if (fromAccount === 'margin' || toAccount === 'margin') {
+            method = 'privatePostMarginTransfer';
+            const symbol = this.safeString (params, 'symbol');
+            if (symbol === undefined) {
+                throw new ExchangeError ('You must define in params what market you are transferring. e.g. {symbol: ETH/BTC}');
+            }
+            const market = this.market (symbol);
+            request['market'] = market['id'];
+            const marginMarkets = await this.publicGetMarginMarket (params);
```

Comment:
At the beginning I thought we could avoid this call by asking to pass the account id in the params. However I think this is better as it feels more unified for the user to pass in the margin symbol in params and for us to find the accountId, as I've done this for another `transfer` function too.

This call at the beginning at the beginning I put into `fetchAccounts` so it could be in the cache with `this.loadAccounts` however it seemed a bit to hard to parse to current `fetchAccounts` structure.

Reply:
> This call at the beginning at the beginning I put into `fetchAccounts` so it could be in the cache with `this.loadAccounts` however it seemed a bit to hard to parse to current `fetchAccounts` structure.

How does that response structure of fetchAccounts look like?

Reply:
So fetchAccounts should return something like: 
```
[
    {
        id: "s32kj302lasli3930",
        type: "main",
        currency: "USDT",
        info: { ... }
    },
```

Coinex when calling `publicGetMarginMarket` returns:
```
{
   "BTCUSDT": 1,    # BTCUSDT:market, 1:account ID
   "BCHUSDT": 2,
   ......
 },
 ```
 And it also has subAccounts that can be found with `privateGetSubAccountBalance` that returns:
 ```
 data: {
    newsubaccount: {
      ZIL: { available: '0', frozen: '0' },
      BAKE: { available: '0', frozen: '0' },
      ...
      },
      ...
  }
```
Also I thought maybe it makes sense during fetchMarkets to get the accountId of the margin symbols, and store it in the market? And keep `fetchAccounts` for the subaccounts in this case?

Reply:
After reading the comment below I'll move `publicGetMarginMarket` to `fetchMarkets`